client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -239,10 +241,15 @@ public RssShuffleManager(SparkConf conf, boolean isDriver) {
     this.rssResubmitStage =
         rssConf.getBoolean(RssClientConfig.RSS_RESUBMIT_STAGE, false)
             && RssSparkShuffleUtils.isStageResubmitSupported();
+    this.taskFailureRetryEnabled =
+        sparkConf.getBoolean(
+            RssSparkConfig.SPARK_RSS_CONFIG_PREFIX
+                + RssClientConf.RSS_TASK_FAILURE_RETRY_ENABLED.key(),
+            RssClientConf.RSS_TASK_FAILURE_RETRY_ENABLED.defaultValue());
```

Comment:
And here you can introduce the new var to indicate whether we need to start shuffle manager rpc service.Like this:

`this.shuffleManagerRpcServiceEnabled = taskBlockSendFailureRetry || rssResubmitStage
`